mkr-mcd.md

```diff
@@ -148,9 +148,54 @@ This allows us to enforce properties after each step, and restore the old state
          <ward> ... ADDR |-> (_ => false) ... </ward>
 ```
 
-`Vat.wish ADDRFROM ADDRTO` checks that `ADDRFROM` has granted control to `ADDRTO`.
-`Vat.hope ADDRTO` and `Vat.nope ADDRTO` set and unset `<can>` for `ADDRTO` from `ADDRFROM`.
-**TODO**: Should we assume that each `ADDRTO` already has `<can>` initialized, or inizialize it here if not?
+### Deactivation
+
+-   `Vat.cage` disables access to this instance of MCD.
+
+**TODO**: Should be `note`.
+
+```k
+    syntax VatAuthStep ::= "cage"
+ // -----------------------------
+    rule <k> Vat . cage => . ... </k>
+         <live> _ => false </live>
+```
+
+### Vat Safety Checks
+
+Vat safety is enforced by adding specific checks on the `<vat>` state updates.
+
+-   `Vat.invariant` states basic invariants of the `Vat` quanities and is checked after every `VatStep`.
+
+```k
+    syntax VatStep ::= "invariant"
+ // ------------------------------
+    rule <k> Vat . invariant => Vat . exception ... </k> [owise]
+    rule <k> Vat . invariant => .               ... </k>
+         <debt> DEBT:Int </debt>
+         <Line> LINE:Int </Line>
+         <vice> VICE:Int </vice>
+         <dai>  DAI      </dai>
+         <sin>  SIN      </sin>
+      requires DEBT >=Int 0 andBool DEBT <=Int LINE
```

Comment:
More accurate to say "No change to Debt would cause DEBT > LINE, still possible for LINE to change and make DEBT > LINE". Better to compare to previous Debt, and only enforce this check on new debt if there is an increase in the debt (but no change to line).

Reply:
Only authorized Vat calls (eg fold) are allowed to violate the LINE/DEBT relationship. Note though, that `drip` from `Jug` is non-auth, but it calls `fold` in an auth manner.